presto-main/src/main/java/com/facebook/presto/operator/scalar/JsonFunctions.java

```diff
@@ -88,38 +90,40 @@ public static JsonPath castCharToJsonPath(@LiteralParameter("x") Long charLength
         return new JsonPath(padSpaces(pattern, charLength.intValue()).toStringUtf8());
     }
 
-    @SqlNullable
     @ScalarFunction("is_json_scalar")
     @LiteralParameters("x")
     @SqlType(StandardTypes.BOOLEAN)
-    public static Boolean varcharIsJsonScalar(@SqlType("varchar(x)") Slice json)
+    public static boolean varcharIsJsonScalar(@SqlType("varchar(x)") Slice json)
     {
         return isJsonScalar(json);
     }
 
-    @SqlNullable
     @ScalarFunction
     @SqlType(StandardTypes.BOOLEAN)
-    public static Boolean isJsonScalar(@SqlType(StandardTypes.JSON) Slice json)
+    public static boolean isJsonScalar(@SqlType(StandardTypes.JSON) Slice json)
     {
         try (JsonParser parser = createJsonParser(JSON_FACTORY, json)) {
             JsonToken nextToken = parser.nextToken();
             if (nextToken == null) {
-                return null;
+                throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "Invalid JSON value: " + truncateIfNecessaryForErrorMessage(json));
             }
 
             if (nextToken == START_ARRAY || nextToken == START_OBJECT) {
                 parser.skipChildren();
                 if (parser.nextToken() != null) {
-                    // Invalid JSON: trailing tokens
-                    return null;
+                    throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "Invalid JSON value: " + truncateIfNecessaryForErrorMessage(json));
                 }
                 return false;
             }
-            return parser.nextToken() == null ? true : null;
+
+            if (parser.nextToken() != null) {
+                // check no trailing token
```

Comment:
This should be above the `if`, since that is what is doing the check. Or change the comment to state the problem:
```java
// no trailing token

Reply:
I personally prefer to put the comment in `if` to explain why we want to do this. Switch to use `no trailing token`. Thank you for the comments!